README.md

```diff
@@ -1,94 +1,235 @@
+    description: "Install Whalesay"
+    pull:
+      name: docker/whalesay
+      tag: latest
+````
+
+### Docker push
+
+#### Syntax
+```yaml
+- docker:
+    description: "Description of the command"
+    push:
+      name: IMAGE_NAME
+      tag: IMAGE_TAG
+      arguments:
+      - arg1
+      - arg2
+      flags:
+        FLAGNAME: FLAGVALUE
+        REPEATED_FLAG:
+        - FLAGVALUE1
+        - FLAGVALUE2
+      outputs:
+        - name: NAME
+          hostname: HOSTNAME
+```
+
+#### Example
+````yaml
+- docker:
+    description: "Push image"
+    push:
+      name: gmadhok/cookies
+      tag: v1.0
+````
+
+### Docker build
+
+#### Syntax
+```yaml
+- docker:
+    description: "Description of the command"
+    build:
+      tag: IMAGE_TAG
+      file: Dockerfile #OPTIONAL
+      path: PATH #defaults to "." OPTIONAL
+      arguments:
+      - arg1
+      - arg2
+      flags:
+        FLAGNAME: FLAGVALUE
+        REPEATED_FLAG:
+        - FLAGVALUE1
+        - FLAGVALUE2
+      outputs:
+        - name: NAME
+          hostname: HOSTNAME
+```
+
+#### Example
+````yaml
+- docker:
+    description: "Build image"
+    build:
+      tag: "gmadhok/cookies:v1.0"
+      file: Dockerfile
+````
+
+### Docker run
+
+#### Syntax
+```yaml
+- docker:
+    description: "Description of the command"
+    run:
+      image: IMAGE
+      name: NAME 
+      detach: BOOL #defaults to false
+      ports:
+        - host: HOST_PORT
+          container: CONTAINER_PORT
+      env:
+        variable: VALUE
+      privileged: BOOL #defaults to false
+      rm: BOOL #defaults to false
+      command: COMMAND
+      arguments:
+      - arg1
+      - arg2
+      flags:
+        FLAGNAME: FLAGVALUE
+        REPEATED_FLAG:
+        - FLAGVALUE1
+        - FLAGVALUE2
+      outputs:
+        - name: NAME
+          hostname: HOSTNAME
+```
+
+#### Example
+```yaml
+ - docker:
+    description: "Run Whalesay"
+    run:
+      name: mixinpractice
+      image: "docker/whalesay:latest"
+      detach: true
+      ports:
+        - host: 8080
+          container: 80
+      env:
+        myvar: "whales"
+      privileged: true
+      rm: true
+      command: cowsay
+      arguments:
+         - "Hello World"
+```
+
+### Docker remove
+
+#### Syntax
+```yaml
+- docker:
+    description: "Description of the command"
+    remove:
+      container: CONTAINER_NAME
+      force: BOOL #defaults to false
+      arguments:
+      - arg1
+      - arg2
+      flags:
+        FLAGNAME: FLAGVALUE
+        REPEATED_FLAG:
+        - FLAGVALUE1
+        - FLAGVALUE2
+      outputs:
+        - name: NAME
+          hostname: HOSTNAME
+```
+
+#### Example
+```yaml
+- docker:
+    description: "Remove mixinpractice"
+    remove:
+      container: mixinpractice
+      force: true
+```
+
+### Docker login
+
+#### Syntax
+Username and password are optional because the mixin will default to using environment variables provided by DOCKER_USERNAME and DOCKER_PASSWORD from a parameter or a credential.
```

Comment:
```suggestion
Username and password are optional because the mixin will default to using environment variables provided by DOCKER_USERNAME and DOCKER_PASSWORD from a parameter or a credential.  See an [example](/examples/docker-mixin-test/README.md#notes-on-docker-login) for how to use 
docker login and securely provide your username and password.
```